kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -219,6 +222,29 @@ translatePredicate translateUninterpreted predicate =
       where
         applicationChildrenSorts = termLikeSort <$> applicationChildren
 
+    translatePredicateExists Exists { existsVariable, existsChild } =
+        case  getHook of
+            Just builtinSort
+              | builtinSort == Builtin.Bool.sort
+              -> translateExists SMT.tBool varTerm existsChild
+              | builtinSort == Builtin.Int.sort
+              -> translateExists SMT.tInt varTerm existsChild
+            _ -> empty
+            
+      where
+        varTerm = mkElemVar existsVariable
+        varSort = sortedVariableSort (getElementVariable existsVariable)
+        tools :: SmtMetadataTools Attribute.Symbol
+        tools = given
+        Attribute.Sort { hook = Hook { getHook } } =
+            sortAttributes tools varSort
+
+    translateExists varSort varTerm predTerm
+      = do
+        var <- translateUninterpreted varSort varTerm
```

Comment:
I'm a little uncertain about this; doesn't `translateUninterpreted` declare a global constant in the SMT-LIB translation?

Reply:
Yes, it does. And in this case it's useless. However, it seems to play well with the fact that this is the regular way to translate variables. If we want to translate them in a different way, we would need to change the way we translate patterns completely, to keep track of free and bound variables. Do you want me to try and do that?

Reply:
@ttuegel I've addressed your concerns by adding an extra argument to `translateUninterpreted` to select whether the variable should be declared or not.  Please take another look.